### PR TITLE
albertoeks/c01-aws02

### PR DESCRIPTION
**Pull request by albertoeks**
# Description

Exercise (e.g. C01-E01): c01-aws02

Student Name: Alberto Santos

> Answers for aws exercise (c01-aws02).

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [x] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-aws02/albertoeks/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+ubuntu@13.210.113.24$ cat file.txt
+Hello S3! 
+```
+
+- Add a brief description of the challenges you faced:
+```
+Two challenges that I would highlight on this exercise are:
+
+1. An error with invalid json, when I tried to create a role from a json file, but actually the error was that I missed the 'file://' on --assume-role-policy-document argument. For me, the message wasn't quite helpful.
```
Comment (alanlima) [inline classes/01class/exercises/c01-aws02/albertoeks/README.md:139]: Unfortunately, not all error messages from aws cli is clear enough. Fortunately, the aws cli docs usually has some examples to how use the commands.
Well done figuring out by yourself 👌

File: classes/01class/exercises/c01-aws02/albertoeks/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+#As my ec2 instance was a ubuntu one, there was no aws-cli out-of-the-box. So, I had to install aws-cli before move on
+
+ubuntu@13.210.113.24$ sudo snap install aws-cli --classic
+
+ubuntu@13.210.113.24$ aws s3 ls c01-aws02
+2020-07-27 12:40:55         10 file.txt
+
+ubuntu@13.210.113.24$ ls
+file.txt
```
Comment (alanlima) [inline classes/01class/exercises/c01-aws02/albertoeks/README.md:129]: It is missing the command to copy the file into the instance. Could you please add it?

**Review by alanlima — CHANGES_REQUESTED** [review]
Great job @albertoeks

Just a small adjustment, please check the notes 🙂

File: classes/01class/exercises/c01-aws02/albertoeks/README.md (hunk 2 above)
Comment (albertoeks) [reply classes/01class/exercises/c01-aws02/albertoeks/README.md]: Good catch @alanlima! Thanks for your review! I've added a new commit with the missing command. Hope it's ok now. Thanks!

File: classes/01class/exercises/c01-aws02/albertoeks/README.md (hunk 1 above)
Comment (albertoeks) [reply classes/01class/exercises/c01-aws02/albertoeks/README.md:142]: You're definitely right. Thanks!

File: classes/01class/exercises/c01-aws02/albertoeks/README.md (hunk 2 above)
Comment (alanlima) [reply classes/01class/exercises/c01-aws02/albertoeks/README.md]: Thanks, it looks ok ;)

File: classes/01class/exercises/c01-aws02/albertoeks/README.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+#As my ec2 instance was a ubuntu one, there was no aws-cli out-of-the-box. So, I had to install aws-cli before move on
+
+ubuntu@13.210.113.24$ sudo snap install aws-cli --classic
+
+ubuntu@13.210.113.24$ aws s3 ls c01-aws02
+2020-07-27 12:40:55         10 file.txt
+
+ubuntu@13.210.113.24$ aws s3 cp s3://c01-aws02/file.txt .
+download: s3://c01-aws02/file.txt to ./file.txt
```
Comment (alanlima) [inline classes/01class/exercises/c01-aws02/albertoeks/README.md:129]: Thanks for the amend ;)

**Review by alanlima — APPROVED** [review]
LGTM

**Pull request merged**